### PR TITLE
OpenAPI: Refactor module to `responder.ext.openapi`

### PR DESCRIPTION
**Pull request by amotl**
## About
This patch refactors the OpenAPI extension, it is living at `responder.ext.openapi` now.

## Setup
Because Responder extensions are optional, install OpenAPI support like this.
```shell
pip install --upgrade 'responder[openapi]'
```

## Usage
```python
>>> from responder.ext.openapi import OpenAPISchema

>>> OpenAPISchema.
OpenAPISchema.add_schema(      OpenAPISchema.docs_response(   OpenAPISchema.openapi          OpenAPISchema.schema_response(
OpenAPISchema.docs             OpenAPISchema.mro()            OpenAPISchema.schema(          OpenAPISchema.static_url(
```

### CODE REVIEW

**Comment by kennethreitz** [comment]
I like making it an extension, but perhaps we should make the dependencies for this default, as it’s going to be the main reason anyone uses this, i think.

**Comment by amotl** [comment]
Dear Kenneth,

thanks for your reply.

> I like making it an extension [...]

Excellent. Actually, it always has been an extension, living at `responder.ext.schema`. Only a few imports have been pulling that into the main `responder` code base. By removing them, it does not significantly degrade usability, but significantly improves modularization and maintainability.

> [...] but perhaps we should make the dependencies for this default, as it’s going to be the main reason anyone uses this, i think.

At [vasuki](https://pypi.org/project/vasuki/), I've actually not used any OpenAPI features at all, and haven't been aware about them at all.

I think it is absolutely viable to follow the original idea to have OpenAPI support as an extension like intended. I didn't have any plans to detour from this, and this patch thorougly inherits this idea, just makes it kinda complete.

Maybe I can convince you going forward with the idea, to have **all three** modules of CLI, GraphQL, and OpenAPI, living equally well and symmetric at `responder.ext` now, like outlined in the [upcoming change log section for v2.1.0](https://github.com/kennethreitz/responder/blob/changelog-2.1.0/CHANGELOG.md)?

With kind regards,
Andreas.

**Comment by kennethreitz** [comment]
excited about this

File: responder/api.py [hunk 1]
```diff
@@ -110,6 +109,14 @@ def __init__(
...
 
         if openapi or docs_route:
+            try:
+                from .ext.openapi import OpenAPISchema
+            except ImportError as ex:
+                raise ImportError(
+                    "The dependencies for the OpenAPI extension are not installed. "
+                    "Install them using: pip install 'responder[openapi]'"
+                ) from ex
+
```
[the remark this answers is not in the document]
Comment (amotl) [reply responder/api.py:119]: 🤷

**Pull request merged**